src/tests/class_tests/smartpeak/source/Utilities_test.cpp

```diff
@@ -382,52 +382,48 @@ BOOST_AUTO_TEST_CASE(endsWith)
   BOOST_CHECK_EQUAL(Utilities::endsWith(names[4], "does_not_end_with_this", false), false);
 }
 
-BOOST_AUTO_TEST_CASE(getPathnameContent)
+BOOST_AUTO_TEST_CASE(getFolderContents)
 {
-  const std::string pathname = SMARTPEAK_GET_TEST_DATA_PATH("");
-  const std::array<std::vector<std::string>, 4> c = Utilities::getPathnameContent(pathname);
-
-  // number of items in the pathname, taking .gitignore into account
-  BOOST_CHECK_EQUAL(c[0].size(), 44);
-  BOOST_CHECK_EQUAL(c[1].size(), 44);
-  BOOST_CHECK_EQUAL(c[2].size(), 44);
-  BOOST_CHECK_EQUAL(c[3].size(), 44);
-
-  BOOST_CHECK_EQUAL(c[0][0], "170808_Jonathan_yeast_Sacc1_1x_1_FluxTest_1900-01-01_000000.featureXML");
- #ifdef _WIN32
-   // NOTE: depending on the build machine...
-   //BOOST_CHECK_EQUAL(c[1][0], "774620"); // file size
- #else
-  BOOST_CHECK_EQUAL(c[1][0], "761937"); // file size
- #endif
-  BOOST_CHECK_EQUAL(c[2][0], ".featureXML");
-
-  BOOST_CHECK_EQUAL(c[0][43], "workflow_csv_files");
-  BOOST_CHECK_EQUAL(c[1][43], "22"); // number of items within the folder
-  BOOST_CHECK_EQUAL(c[2][43], "Directory");
+  std::filesystem::path pathname = SMARTPEAK_GET_TEST_DATA_PATH("");
+  
+  std::array<std::vector<std::string>, 4> directory_entries;
+  
+  std::tuple<std::string, std::string> order_by_name = std::make_tuple ("name", "ascending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_name);
+  BOOST_CHECK_EQUAL(directory_entries[0][0], "170808_Jonathan_yeast_Sacc1_1x_1_FluxTest_1900-01-01_000000.featureXML");
+  directory_entries[0].clear(); directory_entries[1].clear();
+  directory_entries[2].clear(); directory_entries[3].clear();
+  
+  std::tuple<std::string, std::string> order_by_file_extension = std::make_tuple ("extension", "descending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_file_extension);
+  BOOST_CHECK_EQUAL(directory_entries[2][0], "Directory");
+  BOOST_CHECK_EQUAL(directory_entries[2][directory_entries.size()-1], ".txt");
+  directory_entries[0].clear(); directory_entries[1].clear();
+  directory_entries[2].clear(); directory_entries[3].clear();
+  
+  std::tuple<std::string, std::string> order_by_size = std::make_tuple ("size", "descending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_size);
+  BOOST_CHECK_EQUAL(directory_entries[1][0], "3804256");
```

Comment:
Does this value change on the different operating systems?  As I am sure you noticed, we had to comment this out on Windows previously because it kept changing.

Reply:
The testing here was entirely just for making sure the sorting mechanism is behaving as expected, it'll most likely to yield different results on Windows. I have updated the test with guarding directives so that it won't run on windows.